gcs/deleteFile/deleteFile.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// DeleteFile is a probe that deletes a file from GCS.
+// It tests the file deletion aspect of GCS and ensures that the file
+// is deleted after the deletion request is made.
+//
+// TODOs:
+// TODO(evanSpendlove): Setup labels and metrics within probe
+// TODO(evanSpendlove): Add handling of additional labels
+// TODO(evanSpendlove): Add Start() method to Probe.
+// TODO(evanSpendlove): Add initProbeMetrics() method to Probe.
+// TODO(evanSpendlove): Add runProbeForTarget() method to Probe.
+// TODO(evanSpendlove): Add runProbe() method to Probe.
+
+// Package gcs implements the probes required to monitor a GCS instance.
+package gcs
+
+import (
+	"fmt"
+
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+
+	deletepb "github.com/googleinterns/step224-2020/gcs/deleteFile/proto"
+)
+
+// Probe holds aggregate information about all probe runs, per-target.
+// It also holds the config and options used to initialise the probe.
+type Probe struct {
+	name    string
+	config  *deletepb.DeleteFileProbe
+	targets []string
+	opts    *options.Options
+
+	// probeOpLatency is used to record the latency of a probe operation
+	// per target.
+	probeOpLatency map[string]*metrics.EventMetrics
+	// apiCallLatency is used to record latency values with distinct labels
+	// per API call per target.
+	// Recommended usage: append(apiCallLatency[target][apiCall], newMetric)
+	apiCallLatency map[string]map[string][]*metric.EventMetrics
+	logger         *logger.Logger
+
+	bucket string
+	// CancelFunc cancels this probe and its goroutines.
+	// If this probe needs to be cancelled immediately, call CancelFunc().
+	CancelFunc context.CancelFunc
+}
+
+// Init initializes the probe with the given parameters.
+// This is a required method to implement the Probe interface specified
```

Comment:
I would include a reference to the interface that it implements: "Implements the <package>.<Type> interface."

Reply:
I'm not quite sure what you mean by including a reference to the interface.

There is a `Probe` interface specified in probes/probes.go in Cloudprober.

https://github.com/google/cloudprober/blob/baf63376da5f8b54705ac723f58c5233d8dd68cf/probes/probes.go#L67

This is the interface that is implemented.

Are you suggesting that I add a link to reference this, or a file path?

Reply:
I think you should include a line in the comment on the DeleteFileProbe struct to indicate that it implements the Cloudprober.Probes.Probe interface.